### ISSUE TITLE
win_user is not repeatable

### ISSUE THREAD

**Issue by garymcwilliams** [ansible/ansible#24884]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
win_user

##### ANSIBLE VERSION
```
$ ansible --version
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.6.6 (r266:84292, Aug 18 2016, 12:06:14) [GCC 4.4.7 20120313 (Red Hat 4.4.7-17)]
```

##### CONFIGURATION
n/a

##### OS / ENVIRONMENT
Running on Centos 7, managing Windows Server 2016

##### SUMMARY
win_user can create a user, but running task again in playbook results in error

##### STEPS TO REPRODUCE
A minimal playbook

```yaml
---
# vim: set filetype=ansible ff=unix ts=2 sw=2 ai expandtab :
#
# Playbook to configure the environment

- hosts: createuser
  tasks:
    - name: create user
      run_once: true
      win_user:
        name: gary
        password: 'B0bP4ssw0rd123!^'
        password_never_expires: true
        account_disabled: no
        account_locked: no
        password_expired: no
        state: present
        groups:
          - Administrators
          - Users
```

When I run first time I get:

```python
PLAY [createuser] ************************

TASK [Gathering Facts] *******************
ok: [dsy-demo-mssql02]

TASK [create user] ***********************
changed: [dsy-demo-mssql02]

PLAY RECAP *******************************
dsy-demo-mssql02           : ok=2    changed=1    unreachable=0    failed=0

```
When I rerun the same playbook I get an error

Just trying to work out if there is anything obvious I could be doing wrong.

I have verified that I can login to the created account.

verbose output is not much different:
```python
TASK [create user] *********************************
task path: /root/ansible_windows/createuser-playbook.yml:9
Using module file /usr/lib/python2.6/site-packages/ansible/modules/windows/win_user.ps1
<dsy-demo-mssql02> ESTABLISH WINRM CONNECTION FOR USER: local\\Administrator on PORT 5986 TO dsy-demo-mssql02
EXEC (via pipeline wrapper)
fatal: [dsy-demo-mssql02]: FAILED! => {
    "changed": false,
    "failed": true,
    "msg": "Exception calling \"ValidateCredentials\" with \"2\" argument(s): \"The network path was not found.\r\n\""
}
```

##### EXPECTED RESULTS
the play should return 'ok' as the user already exists

##### ACTUAL RESULTS
```python
PLAY [createuser] ************************

TASK [Gathering Facts] *******************
ok: [dsy-demo-mssql02]

TASK [create user] ***********************
fatal: [dsy-demo-mssql02]: FAILED! => {"changed": false, "failed": true, "msg": "Exception calling \"ValidateCredentials\" with \"2\" argument(s): \"The network path was not found.\r\n\""}

NO MORE HOSTS LEFT ***********************
        to retry, use: --limit @/root/ansible_windows/createuser-playbook.retry

PLAY RECAP *******************************
dsy-demo-mssql02           : ok=1    changed=0    unreachable=0    failed=1

```

**Comment by ansibot** [comment]
cc @nitzmahone
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
This works for me on my host but I think I know what the issue is I just can't seem to replicate it. Are you able to edit `/usr/lib/python2.6/site-packages/ansible/modules/windows/win_user.ps1` and place the line `Fail-Json @{} "$host_name"` on line 110. I have a feeling because your hostname is over 15 characters it is not getting the right value with this command. This way we can see if `$host_name` is being set properly.

**Comment by garymcwilliams** [comment]
Changed:
```
            Else {
Fail-Json @{} "This host=$host_name"
                $password_match = $pc.ValidateCredentials($username, $password)
            }
```
results in

```
TASK [create local OS user] *********************************************************************************************************************************************************************************************************
fatal: [dsy-demo-mssql02]: FAILED! => {"changed": false, "failed": true, "msg": "This host=dsy-demo-sql02"}

```

NOTE: the name is "different" (mssql01 => sql02) but other servers have matching names, with the same issue.

So, now that I've found that powrshell script (I really should have been able to work that out), I've had a play on the server with the snippets of code from the script.
(My main problem here is knowing just about nothing of PowerShell, but it's likely useful to know a bit).
```
Add-Type -AssemblyName System.DirectoryServices.AccountManagement
$host_name = [System.Net.Dns]::GetHostName()
$pc = New-Object -TypeName System.DirectoryServices.AccountManagement.PrincipalContext 'Machine', $hostname
$password_match = $pc.ValidateCredentials("lepdm", "fred")

```
results in

```
Exception calling "ValidateCredentials" with "2" argument(s): "The network path was not found.
"
At line:1 char:1
+ $password_match = $pc.ValidateCredentials("lepdm", "fred")
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : FileNotFoundException

```

So, I assume it's something screwy on the windows server, but no real idea what. I don't control the setup of these windows servers, I am just given it, so don't know how the windows images are being built.

**Comment by jborean93** [comment]
Thanks for the info we are getting somewhere and hopefully we find something that fixes your issue. There are 2 things you can try

* remove `$host_name` when defining the PrincipalContext as having that omitted should result in it using localhos
* use `env:$ComputerName` instead of `[System.Net.Dns]::GetHostName()` when setting `$host_name` to see if that works.

Hopefully one of those options work or I find a way to replicate it. While your doing this can you send through some info of your host name as shown by the System settings as well as any network configuration like IPv4 settings and stuff like that.

**Comment by garymcwilliams** [comment]
```
Add-Type -AssemblyName System.DirectoryServices.AccountManagement
$host_name = $ENV:ComputerName
$pc = New-Object -TypeName System.DirectoryServices.AccountManagement.PrincipalContext 'Machine', $hostname
$password_match = $pc.ValidateCredentials("lepdm", "fred")

```
This now gives:
```
PS C:\Users\Administrator> $password_match = $pc.ValidateCredentials("lepdm", "fred")
Exception calling "ValidateCredentials" with "2" argument(s): "Unknown error (0x{0})80075000"
At line:1 char:1
+ $password_match = $pc.ValidateCredentials("lepdm", "fred")
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : PrincipalOperationException

```
which looks like a permission issue, but as I'm local admin I'm not sure why. I'll check that again.
Same error whether password is correct or not.
If I don't include $host_name in the $pc line, then I revert to the Network Path issue.

I have been able to build out my own test windows server and all is fine, it really looks like something is not setup correctly on the generated image(s) that I have been given. 

If you feel able to continue help, then I would appreciate that. I also accept that this looks to be less of a core ansible issue and more an environmental one, so if you want to drop it that's fine.

For now, I have one playbook that I use to create the users I need and I just manually remember which hosts I need to run it on, so it's less than an ideal managed environment, but user creation is only needed on a new setup and that's a rare occurrence (for now anyway).

**Comment by garymcwilliams** [comment]
should also say, I did change the ps1 file on ansible-control:
```
            $host_name = $ENV:ComputerName
            $pc = New-Object -TypeName System.DirectoryServices.AccountManagement.PrincipalContext 'Machine', $host_name

```
when I run this as a playbook though, I get the "Network path" error rather than the "unknown error"

**Comment by jborean93** [comment]
This is a tricky one without being able to replicate it myself there's not too much I can do. There might be more info in the event logs that could possibly tell you more but some things you can try

* Update the version of .net installed as I've seen a few errors with `PrincipalContext` and older versions, just not this exact issue
* If it works on another box try and find differences with how they are set up and try to either break your working one or fix the bad ones.

Hopefully either the logs will give you a bit more info or you can find the issue.

**Comment by ansibot** [comment]
cc @angstwad @cchurch
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by drfickle** [comment]
I'm also hitting this issue. I added the 
```Fail-Json @{} "This host=$host_name"```
line and the hostname did match, even though the it was over 15 characters.

So I took out the host_name argument from
```$pc = New-Object -TypeName System.DirectoryServices.AccountManagement.PrincipalContext 'Machine', $host_name```
and that did work, or at least didn't fail (I was setting it to the same password using update_password=always).

This machine is not overly locked down or customized.

**Comment by jhawkesworth** [comment]
At 19:00 UTC on Tuesday 19 September 2017 this ticket will be processed during the second Windows Sprint,
https://github.com/ansible/community/wiki/Windows:-sprints
we would appreciate if you could join the Sprint to help out with understanding/fixing/closing this issue.

**Comment by dagwieers** [comment]
@garymcwilliams @drfickle We discussed this issue and decided we first need to find a way to reproduce this issue before we can attempt to fix it. I am unable to reproduce this on Windows 2012 R2. Any help with being able to reproduce this is appreciated.

Also, we plan to improve the documentation related to privilege escalation, WinRM restrictions and alternative methods in order to make it easier for users to test if their issues are related to privileges or permissions. A troubleshooting section would be very welcome.

needs_info

**Comment by fyyc1204** [comment]
I'm also hitting this issue

**Comment by jborean93** [comment]
@fyyc1204 are there any similarities with the setup of the initial issue and your environment. We were never able to replicate this issue so we are not sure what the exact problem is. If you can give us the following information it would be really helpful

* OS Version
* PowerShell Version
* Is it in a domain environment or not
* The task you are running
* The full error message (run with `-vvv`) when running the task
* Does it work on any other hosts
* What Ansible version are you on
* The hostname and netbios name
* What transport auth are you using, basic, ntlm, kerberos, credssp
* Are you using a local or domain account to run this module
* Is there anything in the event logs you can see that is related to this error

**Comment by fyyc1204** [comment]
OS Version ： windows server 2008 
PowerShell Version :  5.0.10586.117
Is it in a domain environment or not : not
The task you are running :  ansible windows -m win_user -a "name=aa password=123"
The full error message (run with -vvv) when running the task : msg": "Exception calling \"ValidateCredentials\" with \"2\" argument(s): \"找不到网络路径。\r\n\""
Does it work on any other hosts : NO
What Ansible version are you on : 2.3.2.0

**Comment by jborean93** [comment]
Are you sure about the OS version, are you on Server 2008 R2 and not 2008 as 2008 only supports PowerShell 1-3 and not 5.

Are all the other hosts the same in their setup as this one, are you able to share any of the other information I asked for?

**Comment by fyyc1204** [comment]
OS : 2008 R2
I test on OS version win10 , it is OK

**Comment by fyyc1204** [comment]
在执行成功的主机上执行命令：
```
PS C:\Users\DELL> $pc.ValidateCredentials("aa", "111111")
False
```
在执行失败的主机上执行命令：
```
PS C:\Users\Administrator> $pc.ValidateCredentials("aa","1111111")
Exception calling "ValidateCredentials" with "2" argument(s): "找不到网络路径。
"
At line:1 char:1
+ $pc.ValidateCredentials("aa","1111111")
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : FileNotFoundException
```

**Comment by jborean93** [comment]
That's good info to know, the fact that it doesn't run when doing it locally means it isn't some weird WinRM issue and narrows it down even more. Are there any entries in the Application, System, or Security event log that might indicate what is going wrong here?

**Comment by fyyc1204** [comment]
no

**Comment by fyyc1204** [comment]
但是， 有这样一条错误很可疑
日志名称:          Application
来源:            Microsoft-Windows-PerfNet
日期:            2018/1/4 15:18:35
事件 ID:         2004
任务类别:          无
级别:            错误
关键字:           经典
用户:            暂缺
计算机:           DEVFORZKB.yfgt.com
描述:
无法打开服务器服务性能对象。数据段的第一个四字节 (DWORD) 包含状态代码。
事件 Xml:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Microsoft-Windows-PerfNet" Guid="{CAB2B8A5-49B9-4EEC-B1B0-FAC21DA05A3B}" EventSourceName="PerfNet" />
    <EventID Qualifiers="49152">2004</EventID>
    <Version>0</Version>
    <Level>2</Level>
    <Task>0</Task>
    <Opcode>0</Opcode>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2018-01-04T07:18:35.000000000Z" />
    <EventRecordID>26511</EventRecordID>
    <Correlation />
    <Execution ProcessID="0" ThreadID="0" />
    <Channel>Application</Channel>
    <Computer>DEVFORZKB.yfgt.com</Computer>
    <Security />
  </System>
  <EventData>
    <Binary>340000C0</Binary>
  </EventData>
</Event>

**Comment by fyyc1204** [comment]
- <Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="Microsoft-Windows-PerfNet" Guid="{CAB2B8A5-49B9-4EEC-B1B0-FAC21DA05A3B}" EventSourceName="PerfNet" /> 
  <EventID Qualifiers="49152">2004</EventID> 
  <Version>0</Version> 
  <Level>2</Level> 
  <Task>0</Task> 
  <Opcode>0</Opcode> 
  <Keywords>0x80000000000000</Keywords> 
  <TimeCreated SystemTime="2018-01-04T07:18:35.000000000Z" /> 
  <EventRecordID>26511</EventRecordID> 
  <Correlation /> 
  <Execution ProcessID="0" ThreadID="0" /> 
  <Channel>Application</Channel> 
  <Computer>DEVFORZKB.yfgt.com</Computer> 
  <Security /> 
  </System>
- <EventData>
  <Binary>340000C0</Binary> 
  </EventData>
  </Event>

**Comment by ansibot** [comment]
@garymcwilliams This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@garymcwilliams This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@garymcwilliams You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by fyyc1204** [comment]
@jborean93 PS C:\Users\Administrator> $pc


ContextType     : Machine
Name            : RABBITMQ
Container       :
UserName        :
Options         : Negotiate
ConnectedServer :

**Comment by fyyc1204** [comment]
ConnectedServer :   is  NULL

**Comment by fyyc1204** [comment]
目前发现出现这个问题有两种原因：
1、server 服务没有启动
2、网卡驱动有问题  (执行 ipconfig /renew  验证 )  ：如果有错误提示，参考下面链接解决  https://answers.microsoft.com/en-us/windows/forum/windows_vista-networking/an-error-occurred-while-releasing-interface/4261cacb-e924-4d74-a349-4f3b898c27a9

**Comment by fyyc1204** [comment]
还有一种情况是虚拟机上有多个网卡，删除不用的网卡就解决了

**Comment by wsgzao** [comment]
@fyyc1204 @jborean93 
Hello Everyone, I also use ansible 2.4.3.0 with powshell 3.0 deal windows7/2008/2012 and met the same problem below

Exception calling \"ValidateCredentials\" with \"2\" argument(s): 

Just start the service "Server" that will solve the problem

**Comment by ansibot** [comment]
@garymcwilliams You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by cawilliamson** [comment]
Hi folks - hijacking this ticket (rather than creating a duplicate) as I am facing the same issue.

The solution recommended from @wsgazo didn't resolve the issue for me sadly - the "Server" service was already running and restarting it had no effect.

Any progress on this?

**Comment by ansibot** [comment]
@garymcwilliams You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by renatoram** [comment]
Hello, 
I'm hitting the same problem (and it will soon be a problem in production).

OS: Windows 10 Enterprise
Version: 10.0.14393
SMB services completely off for security reasons.

Ansible Version: 2.3.0.0 (but the ansible version seems to be irrelevant here).

The issue is local to the Windows machine. It's not in a Domain environment, and it's using the Machine Principal context.

Example code that fails consistently:

```
$LocalUser = "test1"
$LocalPass = "Pass.123456789"

    Function Test-LocalAuth {
        param($Username, $Password)
        Add-Type -AssemblyName System.DirectoryServices.AccountManagement
        $Obj = New-Object System.DirectoryServices.AccountManagement.PrincipalContext('machine', $env:COMPUTERNAME)
        $Compliance = $Obj.ValidateCredentials($Username, $Password)
        $Compliance
    }

Test-LocalAuth -Username $LocalUser -Password $LocalPass
```

The output is
```
Exception calling "ValidateCredentials" with "2" argument(s): "The network path was not found.
"
At C:\Users\RMDAdmin\Documents\test_credentialvalidation.ps1:9 char:9
+         $Compliance = $Obj.ValidateCredentials($Username, $Password)
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : PrincipalOperationException
```

Our hostname is 14 characters (with a '-' in it: I suspect this might be part of the problem?)

I know it's not the same exact code from win_user.ps1, but this fails the same exact way (and it's taken directly from the ansible module).

```
[void][system.reflection.assembly]::LoadWithPartialName('System.DirectoryServices.AccountManagent')
$host_name = [System.Net.Dns]::GetHostName()
$pc = New-Object -TypeName System.DirectoryServices.AccountManagement.PrincipalContext 'Machine', $host_name
$pc.ValidateCredentials('test1', 'Pass.123456789')
```

My "Server" service being up or not seems to be irrelevant, btw (it's one of the suggested "solutions" I found scouring MS forums).

Can you please reopen this? At the moment, the ansible module is basically unusable for us.

**Comment by jborean93** [comment]
@renatoram there's not much we can really do without understanding what is the actual cause of the issue. Right now I have never come across this issue and without a solid reproducer it is quite hard for us to find the root cause. If you ever find out what is wrong then please let us know.

**Comment by ansibot** [comment]
@garymcwilliams You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
bot_skip

**Comment by forquare** [comment]
**OS/Environment:**
ansible 2.6.1 (installed via pkg(8)) running on FreeBSD 11.2, managing Windows 10 Pro 64-bit running on VMware ESXi.

**Windows Computer name:** win-jenkins-test

**PowerShell version:** 5.1.17134.164 (where: major.minor.build.revision)

**DotNet:**
Both 3.5.20160716 and 4.7.2053.0 via chocolatey

Our connection configuration looks like this:
```
ansible_user: autosupport
ansible_password: superSecretPass
ansible_port: 5985
ansible_connection: winrm
ansible_winrm_scheme: http
ansible_winrm_server_cert_validation: ignore
ansible_become_method: runas
ansible_become_user: "{{ ansible_user }}"
ansible_become_password: "{{ ansible_password }}"
```

Snippet from Playbook:
```
- name: "[JENKINS][WINDOWS] - Create Jenkins user"
  win_user:
    name: "{{ jenkins_user }}"
    password: "{{ jenkins_user_password }}"
    password_expired: no
    password_never_expires: yes
    fullname: "{{ jenkins_user }}"
    description: "CI User"
    state: present
    user_cannot_change_password: yes
    groups:
      - Administrators
  register: jenkins_user_output
```
Doesn't seem to matter if we use `become` or not, this still fails:
```
fatal: [win-jenkins-test]: FAILED! => {
    "changed": false, 
    "msg": "Exception calling \"ValidateCredentials\" with \"2\" argument(s): \"The network path was not found.\r\n\""
}
```

Running setup against the Windows machine in question:
[windows_setup.txt](https://github.com/ansible/ansible/files/2208866/windows_setup.txt)
Snippet from Events log:
[wun_user_issue_events.txt](https://github.com/ansible/ansible/files/2209089/wun_user_issue_events.txt)

**Comment by renatoram** [comment]
@jborean93 I have an update and a probable cause (and solution, for me)

The problem is that "File and Printer Sharing Service" is turned off on all interfaces, *and* the Server service is down (or both).

**Steps to reproduce:**

1. On all network interfaces, remove the "File and Printer Sharing" feature
2. Restart the Server service (LanmanServer)
3. Reboot the machine

The win_user module will fail with this error.

**How to check:**

`netstat -an | findstr 445`

If there is no port 445 listening, the module (and the ValidateCredentials command underneath) will fail.

**How to fix**
1. Turn on "File and Printer Sharing" on at least one interface. This will probably work with a dummy additonal Loopback interface.
2. Restart Server service (and set it to autostart)

Now check if the port is open: the starting mechanism of the SMB2 server seems iffy, if it's not there, reboot the machine. In the same vein, removing the Sharing feature from all NICs and then restarting the Server service does **not** reliably shut down port 445 until you reboot.

**Comment by jborean93** [comment]
Thanks @renatoram for looking at that and finding the issue. Sounds like we need to move away from a solution that relies on RPC/SMB as we can see it is not always available.

**Comment by forquare** [comment]
On the machine I am seeing this on, "File and Printer Sharing" is turned on and listening on 0.0.0.0:445.  Firewalls are turned off.  Rebooting the machine, nor restarting the Server service seems to fix it for me.

**Comment by renatoram** [comment]
@jborean93 I briefly looked into it and according to MS documentation there doesn't seem to be many alternatives. There's a ["LogonUser" API](https://support.microsoft.com/en-ca/help/180548/how-to-validate-user-credentials-on-microsoft-operating-systems) but I'm no coder (especially no Windows coder) so I'm not able to test it.

@forquare also check the status of the SMB services: there are several variables involved I'm afraid. 

In a powershell terminal use `Get-SmbServerConfiguration` and check for SMB2.

You should be able to enable/disable it with 
`Set-SmbServerConfiguration –EnableSMB2Protocol $true` (or $false)

...unless it's yet another different cause.
Did you try my short PS script from [this comment](https://github.com/ansible/ansible/issues/24884#issuecomment-401707071) ? Do you get the same "Network path was not found" error?

Also, double-check that using `Get-NetAdapterBinding` you see "ms_server" as True ([further reading](http://techgenix.com/using-powershell-disable-network-adapter-bindings/))

**Comment by jborean93** [comment]
@renatoram yea I have a feeling our choices are going to be quite limited. We already have some examples of using LogonUser through P/Invoke so it is entirely possible. Will have to play around and see what we can come up with as `ValidateCredentials` seem to be quite brittle as evident by this thread.

**Comment by forquare** [comment]
@renatoram thanks for your reply.

```
PS C:\Users\autosupport> Get-SmbServerConfiguration

AnnounceComment                 :
AnnounceServer                  : False
AsynchronousCredits             : 64
AuditSmb1Access                 : False
AutoDisconnectTimeout           : 15
AutoShareServer                 : True
AutoShareWorkstation            : True
CachedOpenLimit                 : 10
DurableHandleV2TimeoutInSeconds : 180
EnableAuthenticateUserSharing   : False
EnableDownlevelTimewarp         : False
EnableForcedLogoff              : True
EnableLeasing                   : True
EnableMultiChannel              : True
EnableOplocks                   : True
EnableSecuritySignature         : False
EnableSMB1Protocol              : False
EnableSMB2Protocol              : True  <---
EnableStrictNameChecking        : True
EncryptData                     : False
IrpStackSize                    : 15
KeepAliveTime                   : 2
MaxChannelPerSession            : 32
MaxMpxCount                     : 50
MaxSessionPerConnection         : 16384
MaxThreadsPerQueue              : 20
MaxWorkItems                    : 1
NullSessionPipes                :
NullSessionShares               :
OplockBreakWait                 : 35
PendingClientTimeoutInSeconds   : 120
RejectUnencryptedAccess         : True
RequireSecuritySignature        : False
ServerHidden                    : True
Smb2CreditsMax                  : 2048
Smb2CreditsMin                  : 128
SmbServerNameHardeningLevel     : 0
TreatHostAsStableStorage        : False
ValidateAliasNotCircular        : True
ValidateShareScope              : True
ValidateShareScopeNotAliased    : True
ValidateTargetName              : True
```
I'm guessing this means that SMB is enabled?

```
PS C:\WINDOWS\system32> Get-NetAdapterBinding

Name                           DisplayName                                        ComponentID          Enabled
----                           -----------                                        -----------          -------
Ethernet                       File and Printer Sharing for Microsoft Networks    ms_server            True
Ethernet                       Microsoft LLDP Protocol Driver                     ms_lldp              True
Ethernet                       Microsoft Network Adapter Multiplexor Protocol     ms_implat            False
Ethernet                       Link-Layer Topology Discovery Mapper I/O Driver    ms_lltdio            True
Ethernet                       Internet Protocol Version 6 (TCP/IPv6)             ms_tcpip6            True
Ethernet                       Internet Protocol Version 4 (TCP/IPv4)             ms_tcpip             True
Ethernet                       Link-Layer Topology Discovery Responder            ms_rspndr            True
Ethernet                       Client for Microsoft Networks                      ms_msclient          True
Ethernet                       QoS Packet Scheduler                               ms_pacer             True
```
I placed your PS script into a filed called test.ps1 and ran it:
```
PS C:\Users\autosupport\Desktop> .\test.ps1
False
```

**Comment by renatoram** [comment]
So, the script completes with no error (False just means the user is not there or the password is wrong, which is probably correct if you used the script as-is without putting in one of your own users), which is consistent with my testing. If you change the two variables at the top of the script with one of your valid users and passwords do you get True?

On my machines the script exited with the Exception I pasted in my old comment.

Apparently it's breaking elsewhere?

My few lines of PS should be equivalent to using the module, but @jborean93 is better suited than me to confirm it.

**Comment by forquare** [comment]
@renatoram very odd.  I had already replaced your sample credentials with the account here, I've double checked and my credentials are indeed still there.  Running the script still returns False...

This is way above me, so aside from running stuff I'm afraid I don't really know what to do

**Comment by jborean93** [comment]
@forquare @renatoram, I haven't used the module in a while but IIRC it will always returned `changed: True` when `update_password` is `always` so not sure if the ValidateCredentials code path is actually doing anything. https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_user.ps1#L93-L110 indicates a change/password reset should only occur when ValidateCredentials is false and if it is always returning false then maybe we should get rid of it altogether.

**Comment by jborean93** [comment]
@forquare @renatoram (and anyone else interested), I've opened a PR that uses `LogonUser` to validate the credentials and it would be great if you could test it out to see if it fixes your issue. Even if it doesn't I'm hoping it will give us a better (or even different) error that helps to narrow this down even more.

The PR is here https://github.com/ansible/ansible/pull/43059

**Comment by forquare** [comment]
@jborean93 , that seems to have fixed the issue for me.  Thanks very much.
